packages/hebao_v2/contracts/base/libwallet/GuardianLib.sol

```diff
@@ -108,6 +111,48 @@ library GuardianLib
         _removeGuardian(wallet, guardian, 0, true);
     }
 
+    function resetGuardians(
+        Wallet    storage  wallet,
+        address[] calldata newGuardians
+        )
+        external
+    {
+        Guardian[] memory allGuardians = guardians(wallet, true);
+        for (uint i = 0; i < allGuardians.length; i++) {
+            _removeGuardian(wallet, allGuardians[i].addr, GUARDIAN_PENDING_PERIOD, false);
+        }
+
+        for (uint j = 0; j < newGuardians.length; j++) {
+            _addGuardian(wallet, newGuardians[j], GUARDIAN_PENDING_PERIOD, false);
+        }
+    }
+
+    function resetGuardiansWA(
+        Wallet    storage  wallet,
+        bytes32            domainSeperator,
+        Approval  calldata approval,
+        address[] calldata newGuardians
+        )
+        external
+    {
+        wallet.verifyApproval(
+            domainSeperator,
+            SigRequirement.MAJORITY_OWNER_REQUIRED,
+            approval,
+            abi.encode(
+                RESET_GUARDIANS_TYPEHASH,
+                approval.wallet,
+                approval.validUntil,
+                newGuardians
```

Comment:
Needs to be `keccak256(abi.encodePacked(newGuardians))`?

Reply:
Sorry for the delay, fixed.